ctpolicy/ctpolicy.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ctpolicy contains structs describing CT policy requirements and
+// corresponding logic.
+package ctpolicy
```

Comment:
(General) At the moment it feels like there's a bunch of boilerplate and types that aren't really needed; I'd lean towards ruthlessly cutting out as much stuff as possible.  (However, it may well be that you've got future plans for things that currently look extraneous to me, so YMMV)

Reply:
I tried to simplify it. Some yet-unused fields are still there.